tests/test_file_open_gui.py

```diff
@@ -0,0 +1,348 @@
+            process.kill()
+        else:
+            assert process.returncode == 0
+
+
+@pytest.fixture(scope="module")
+def app_dialog(host_application):
+    """
+    Retrieve the application dialog and return the AppDialogAppWrapper.
+    """
+    before = time.time()
+    while before + 30 > time.time():
+        if sgtk.util.is_windows():
+            app_dialog = AppDialogAppWrapper(topwindows)
+        else:
+            app_dialog = AppDialogAppWrapper(topwindows["python"])
+
+        if app_dialog.exists():
+            yield app_dialog
+            app_dialog.close()
+            return
+    else:
+        raise RuntimeError("Timeout waiting for the app dialog to launch.")
+
+
+class AppDialogAppWrapper(object):
+    """
+    Wrapper around the app dialog.
+    """
+
+    def __init__(self, parent):
+        """
+        :param root:
+        """
+        self.root = parent["Shotgun: File Open"].get()
+
+    def exists(self):
+        """
+        ``True`` if the widget was found, ``False`` otherwise.
+        """
+        return self.root.exists()
+
+    def close(self):
+        self.root.buttons["Close"].get().mouseClick()
+
+
+def test_ui_validation(app_dialog, sg_project):
+    """
+    Basic UI validation to make sure all buttons, tabs and fields are available
+    """
+    # Make Sure the File Open dialog is showing up in the right context
+    assert app_dialog.root.captions["File Open"].exists(), "Not the File Open dialog"
+    assert app_dialog.root.captions[
+        "*Project " + sg_project["name"]
+    ].exists(), "Not the right context"
+
+    # Make sure the breadcrumb UI is fine.
+    assert app_dialog.root.buttons[
+        "nav_home_btn"
+    ].exists(), "Nav home button is missing"
+    assert app_dialog.root.buttons[
+        "nav_prev_btn"
+    ].exists(), "Nav previous button is missing"
+    assert app_dialog.root.buttons[
+        "nav_next_btn"
+    ].exists(), "Nav next button is missing"
+    assert app_dialog.root.captions["My Tasks"].exists(), "Not the right breadcrumb"
+
+    # Make sure the all tabs are showing up.
+    assert app_dialog.root.tabs["My Tasks"].exists(), "My Tasks tab is missing"
+    assert app_dialog.root.tabs["Assets"].exists(), "Assets tab is missing"
+    assert app_dialog.root.tabs["Shots"].exists(), "Shots tab is missing"
+    assert app_dialog.root.tabs["All"].exists(), "All tab is missing"
+    assert app_dialog.root.tabs["Working"].exists(), "Working tab is missing"
+    assert app_dialog.root.tabs["Publishes"].exists(), "Publishes tab is missing"
+
+    # Make sure all buttons are showing up
+    assert app_dialog.root.buttons[
+        "+ New Task"
+    ].exists(), "+ New Task button is missing"
+    assert app_dialog.root.buttons[
+        "+ New File"
+    ].exists(), "+ New File button is missing"
+    assert app_dialog.root.buttons["Cancel"].exists(), "Cancel button is missing"
+    assert app_dialog.root.buttons["Open"].exists(), "Open button is missing"
```

Comment:
We discussed setting up hooks to assert whether the open operation was called correctly. 
Did you look into that? This is all good stuff anyway I just wondered if it was possible test that the hook was called with the correct operation.

Reply:
Not for now, this will need us to be able to bootstrap the config to be able to pass `sgtk.sgtk_from_path()` or `sgtk.sgtk_from_path()`
Maybe a good idea for a next innovation day :)